nearcore/src/config_validate.rs

```diff
@@ -223,15 +223,44 @@ impl<'a> ConfigValidator<'a> {
     }
 
     fn validate_cloud_archival_config(&mut self) {
-        if !self.config.archive && self.config.cloud_storage.is_some() {
+        let Some(cloud_archival) = &self.config.cloud_archival else {
+            return;
+        };
+
+        if !self.config.archive {
+            let error_message =
+                "`archive` is false, but `cloud_archival` is configured.".to_string();
+            self.validation_errors.push_config_semantics_error(error_message);
+        }
+        if cloud_archival.is_archival_reader && self.config.cold_store.is_none() {
             let error_message =
-                "`archive` is false, but `cloud_storage` is configured.".to_string();
+                "`cloud_archival` is configured in reader mode, but `cold_store` is missing."
+                    .to_string();
             self.validation_errors.push_config_semantics_error(error_message);
         }
-        // TODO(cloud_archival) Allow for cloud storage independently of cold store.
-        if self.config.cloud_storage.is_some() && self.config.cold_store.is_none() {
+        if !cloud_archival.is_archival_reader && self.config.rpc.is_some() {
+            let error_message =
+                "`cloud_archival` is not configured in reader mode, yet `rpc` is enabled."
+                    .to_string();
+            self.validation_errors.push_config_semantics_error(error_message);
+        }
+        let tracked_shards = self.config.tracked_shards_config();
+        if cloud_archival.is_archival_reader && !tracked_shards.tracks_all_shards() {
```

Comment:
Does reader archival need to track all shards? Will this change after cold store migration to the cloud?

Reply:
I'm ok with this as long as we remember to relax it after single shard tracking 👍